pkg/core/writer/writer.go

```diff
@@ -97,17 +100,23 @@ func New(conf *config.WriterConfig, dpChan chan []*datapoint.Datapoint, eventCha
 
 	ctx, cancel := context.WithCancel(context.Background())
 
-	dimensionClient, err := dimensions.NewDimensionClient(ctx, conf)
-	if err != nil {
-		cancel()
-		return nil, err
+	var dimensionClient *dimensions.DimensionClient
+	var err error
+	if conf.SignalFxAccessToken != "" {
```

Comment:
There are many legitimate configs where the access token is blank but we still want to send (e.g. where the agent sends to a central gateway/proxy that adds the token).  Our Helm chart required it in K8s but what you did to make it optional was appropriate and should be done to make it consistent with the variety of deployment models.

You should conditionalize this based on the presence of `conf.APIUrl`.

Reply:
The helm chart offers to set `APIUrl`, `IngestURL`, etc or rely on `realm`, which has a default value of `us0`. My impression from that is that `APIUrl` will never be empty as result. What's the appropriate move there?

Reply:
They would need to override it to be null in the Helm values they provide to disable SignalFx sending.

Reply:
OK I have changed the helm chart to disable signalfx sending.